### PR TITLE
Fixes #1750 - Allow creation of screenshot file at a custom location

### PR DESCRIPTION
**Pull request by lake-of-dreams**
This PR adds the feature to customize the location where screenshot file can be placed based on the options steps configuration as follows-

* filename: Contains File name only, create_file_in_artifacts: false - Creates screenshot at top level in the source directory
* filename: Contains File name only, create_file_in_artifacts: true - Creates screenshot at top level in the artifatcs directory
* filename: Contains absolute path of file, create_file_in_artifacts: false - Creates screenshot at an absolute path in the box
* filename: Contains relative path of file, create_file_in_artifacts: false - Creates screenshot at an relative path to source directory
* filename: Contains relative path of file, create_file_in_artifacts: true - Creates screenshot at an relative path to the artifacts directory
* filename: Contains absolute path of file, create_file_in_artifacts: true - Creates screenshot at an relative path to the artifacts directory (Leading forward / in the path is ignored)

Tested all scenarios [here](https://app.wercker.com/ravi-r-singhal/screenshot-sample/runs/build/5ab0d4748812290001a52e54)

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -1,4 +1,41 @@
-phantomjs --ignore-ssl-errors=true $WERCKER_STEP_ROOT/screenshot.js $WERCKER_SCREENSHOT_URL \
-    ${WERCKER_REPORT_ARTIFACTS_DIR}/${WERCKER_SCREENSHOT_FILENAME} \
-    $WERCKER_SCREENSHOT_VIEWPORT_WIDTH \
-    $WERCKER_SCREENSHOT_VIEWPORT_HEIGHT
+if [ -z "${WERCKER_SCREENSHOT_FILENAME}" ]
+then
+    echo "\${WERCKER_SCREENSHOT_FILENAME} is empty"
```
Comment (hatchan) [inline run.sh]: This should not contain the envvar, but it should contain the argument that you use in the wercker.yml (users are not going to specify envvar, but they are providing arguments in the wercker.yml).

File: run.sh [hunk 2]
```diff
@@ -1,4 +1,41 @@
...
+else
```
Comment (hatchan) [inline run.sh]: If you add a `exit 1` at the previous `if`, then you can remove this `else` and remove the indentation.

File: run.sh [hunk 3]
```diff
@@ -1,4 +1,41 @@
...
+    FILE_PARENT_PATH=${WERCKER_SCREENSHOT_FILENAME%/*}
```
Comment (hatchan) [inline run.sh]: We should not rely on this notation, as I believe it is part of bash and not sh. Maybe look into commands like `basename`.

File: run.sh [hunk 4]
```diff
@@ -1,4 +1,41 @@
...
+    FILE_NAME=${WERCKER_SCREENSHOT_FILENAME##*/}
+    IS_FILE_PATH_ABSOLUTE="false"
+    if [[ $WERCKER_SCREENSHOT_FILENAME == /* ]]
```
Comment (hatchan) [inline run.sh]: Put `"` around the variable name

File: run.sh [hunk 5]
```diff
@@ -1,4 +1,41 @@
...
+    then
+        IS_FILE_PATH_ABSOLUTE="true"
+    fi
+    if [ "${FILE_PARENT_PATH}" != "${FILE_NAME}" ]
+    then
+        if [ -z "${WERCKER_SCREENSHOT_CREATE_FILE_IN_ARTIFACTS}" ] || [ "false" == "${WERCKER_SCREENSHOT_CREATE_FILE_IN_ARTIFACTS}" ]
+        then
+            if [ "false" == $IS_FILE_PATH_ABSOLUTE ]
```
Comment (hatchan) [inline run.sh]: Put `"` around the variable name

File: run.sh (hunk 1 above)
Comment (lake-of-dreams) [reply run.sh]: Fixed

File: run.sh (hunk 2 above)
Comment (lake-of-dreams) [reply run.sh]: Fixed

File: run.sh (hunk 3 above)
Comment (lake-of-dreams) [reply run.sh]: Verified that works in both sh and bash

File: run.sh (hunk 5 above)
Comment (lake-of-dreams) [reply run.sh]: Fixed

File: run.sh (hunk 4 above)
Comment (lake-of-dreams) [reply run.sh]: Here can't as it is not a string comparision

**Comment by lake-of-dreams** [comment]
Merging as discussed with @hatchan

**Pull request merged**